src/Integration/ModulesIntegration.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Sentry\Integration;
+
+use Composer\Composer;
+use Composer\Factory;
+use Composer\IO\NullIO;
+use Sentry\Event;
+use Sentry\Options;
+use Sentry\State\Hub;
+use Sentry\State\Scope;
+
+/**
+ * This middleware logs with the event details all the versions of the packages
+ * installed with Composer, if any.
+ *
+ * @author Stefano Arlandini <sarlandini@alice.it>
+ */
+final class ModulesIntegration implements IntegrationInterface
+{
+    /**
+     * @var Options The client option
+     */
+    private $options;
+
+    /**
+     * @var array
+     */
+    private static $loadedModules = [];
+
+    /**
+     * Constructor.
+     *
+     * @param Options $options The Raven client configuration
+     */
+    public function __construct(Options $options)
+    {
+        if (!class_exists(Composer::class)) {
+            throw new \LogicException('You need the "composer/composer" package in order to use this middleware.');
+        }
+
+        $this->options = $options;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setupOnce(): void
+    {
+        Scope::addGlobalEventProcessor(function (Event $event) {
+            $self = Hub::getCurrent()->getIntegration($this);
+            if ($self instanceof self) {
+                self::applyToEvent($self, $event);
+            }
+
+            return $event;
+        });
+    }
+
+    /**
+     * @param ModulesIntegration $self
+     * @param Event              $event
+     *
+     * @return null|Event
+     */
+    public static function applyToEvent(self $self, Event $event): ?Event
```

Comment:
Do we really need this (and similar) methods to be static?

Reply:
The reason for this to be static is if there are multiple clients, with multiple `ModulesIntegration` we have `self::$loadedModules` as a cache.